packages/kbn-eslint-plugin-eui-a11y/helpers/get_intent_from_node.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { TSESTree } from '@typescript-eslint/typescript-estree';
+
+/*
+    Attempts to get a string representation of the intent
+    out of an array of nodes.
+    
+    Currently supported node types in the array:
+    * String literal text (JSXText)
+    * Translated text via <FormattedMessage> component -> uses prop `defaultMessage`
+    * Translated text via {i18n.translate} call -> uses passed options object key `defaultMessage` 
+*/
+export function getIntentFromNode(originalNode: TSESTree.JSXOpeningElement): string {
+  const parent = originalNode.parent as TSESTree.JSXElement;
+
+  const node = Array.isArray(parent.children) ? parent.children : [];
+
+  if (node.length === 0) {
+    return '';
+  }
+
+  /*
+    In order to satisfy TS we need to do quite a bit of defensive programming.
+    This is my best attempt at providing the minimum amount of typeguards and
+    keeping the code readable. In the cases where types are explicitly set to
+    variables, it was done to help the compiler when it couldn't infer the type.
+    */
+  return node.reduce((acc: string, currentNode) => {
+    switch (currentNode.type) {
+      case 'JSXText':
+        // When node is a string primitive
+        return `${acc}${currentNode.value} `;
+
+      case 'JSXElement':
+        // Determining whether node is of form `<FormattedMessage defaultMessage="message" />`
+        const name: TSESTree.JSXTagNameExpression = currentNode.openingElement.name;
```

Comment:
NIT should we us a different name like `openingElement` ot `jsxTagObject` or something else so we won't have the `name.name` on[ line 48](https://github.com/elastic/kibana/pull/215859/files#diff-5137898a4b652d250758238bea6b84edb4e4ead45bb64aa86b10bdb40fc5160cR48)

Reply:
Agreed. Will adjust.